go/chat/teams.go

```diff
@@ -364,7 +364,19 @@ func LoadTeam(ctx context.Context, g *libkb.GlobalContext, tlfID chat1.TLFID,
 			return team, err
 		}
 		if !tlfID.EqString(team.KBFSTLFID()) {
-			return team, fmt.Errorf("mismatch TLFID to team: %s != %s", team.KBFSTLFID(), tlfID)
+			return team, ImpteamUpgradeBadteamError{
+				Msg: fmt.Sprintf("mismatch TLFID to team: %s != %s", team.KBFSTLFID(), tlfID),
+			}
+		}
+		impTeamName, err := team.ImplicitTeamDisplayNameString(ctx)
+		if err != nil {
+			return team, err
+		}
+		if impTeamName != tlfName {
```

Comment:
If these names aren't canonicalized (sbs resolves and ordering) they may not align in some cases. There's `teams.ResolveImplicitTeamDisplayName`. It could be you want that, idk.